pkg/orchestrator/gitHubActions.go

```diff
@@ -1,79 +1,116 @@
 package orchestrator
 
 import (
-	"os"
-	"strings"
+	"fmt"
+	piperHttp "github.com/SAP/jenkins-library/pkg/http"
 	"time"
 
 	"github.com/SAP/jenkins-library/pkg/log"
 )
 
-type GitHubActionsConfigProvider struct{}
+type GitHubActionsConfigProvider struct {
+	run run
+}
+
+func (g *GitHubActionsConfigProvider) initOrchestratorProvider(settings *OrchestratorSettings) error {
+	client := piperHttp.Client{}
+	client.SetOptions(piperHttp.ClientOptions{
+		Password:         settings.GitHubToken,
+		MaxRetries:       3,
+		TransportTimeout: time.Second * 10,
+	})
+	ghURL := getEnv("GITHUB_URL", "")
+	switch ghURL {
+	case "https://github.com/":
+		ghURL = "https://api.github.com/"
+	default:
+		ghURL += "api/v3/"
+	}
+	resp, err := client.GetRequest(
+		fmt.Sprintf(
+			"%s/repos/%s/actions/runs/%s", ghURL, getEnv("GITHUB_REPOSITORY", ""), getEnv("GITHUB_RUN_ID", ""),
+		),
+		map[string][]string{
+			"Accept":        {"application/vnd.github+json"},
+			"Authorization": {"Bearer $GITHUB_TOKEN"},
+		}, nil)
+	if err != nil {
+		return fmt.Errorf("can't get API data: %w", err)
 
-func (g *GitHubActionsConfigProvider) InitOrchestratorProvider(settings *OrchestratorSettings) {
-	log.Entry().Debug("Successfully initialized GitHubActions config provider")
+	}
+	err = piperHttp.ParseHTTPResponseBodyJSON(resp, g.run)
+	if err != nil {
+		return fmt.Errorf("can't parse JSON data: %w", err)
+	}
+	return err
 }
 
 func (g *GitHubActionsConfigProvider) OrchestratorVersion() string {
+	log.Entry().Debugf("OrchestratorVersion() for GitHub Actions is not applicable.")
 	return "n/a"
 }
 
 func (g *GitHubActionsConfigProvider) OrchestratorType() string {
-	return "GitHubActions"
+	return "GitHub Actions"
 }
 
 func (g *GitHubActionsConfigProvider) GetBuildStatus() string {
-	log.Entry().Infof("GetBuildStatus() for GitHub Actions not yet implemented.")
-	return "FAILURE"
+	// By default, we will assume it ass a success
+	// On error it would be handled by the action itself
+	return "SUCCESS"
 }
 
 func (g *GitHubActionsConfigProvider) GetLog() ([]byte, error) {
-	log.Entry().Infof("GetLog() for GitHub Actions not yet implemented.")
+	// It's not possible to get log during workflow execution
```

Comment:
I can get the log by:
- Getting `GITHUB_RUN_ID` from the [environment variables](https://docs.github.com/en/actions/learn-github-actions/environment-variables#default-environment-variables)
- Getting the job IDs with ```curl -L \
  -H "Accept: application/vnd.github+json" \
  -H "Authorization: <TOKEN>" \
  <apiURL>/<organisation>/<repository>/actions/runs/<runID>/jobs```
- Logs for every job ID with ```curl -L \
  -H "Accept: application/vnd.github+json" \
  -H "Authorization: Bearer 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" \
  <apiURL>/<organisation>/<repository>/actions/jobs/<jobID>/logs```